src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +18,47 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save phone " + phone + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return phone;
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = cb.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            Predicate resultingPredicate = null;
+            for (String key : params.keySet()) {
+                CriteriaBuilder.In<String> predicate = cb.in(root.get(key));
+                for (String attribute : params.get(key)) {
+                    predicate.value(attribute);
+                }
+                resultingPredicate = resultingPredicate == null ? predicate :
+                        cb.and(resultingPredicate, predicate);
+            }
+            if (resultingPredicate != null) {
```

Comment:
can we write this logic without checking for null?

Reply:
yup, found a default predicate that can be used instead of null